react-js/styling-react.md

```diff
@@ -0,0 +1,250 @@
+
+### Lesson
+
+- [Slides: CSS in React ](https://drive.google.com/open?id=15oauTP4afibqMvNKW5XgVC1qraYUKxY-)
+
+### Guided Practice
+
+> #### Let's create a simple div, and style it using inline CSS and flexbox.  Then we will achieve same using External CSS and Bootstrap separately.
+
+- First thing we need to do is create a react application.
+	- Create a react app using create react app command. `npx create-react-app`.
+    - This will create a simple react app , and now we will start making changes to this app.
+	- First of all get rid of all the unnecessary stuff showing on DOM , remove all the lines in render of `App.js`. 
+	- Now we create a simple div **parent div** which will contain few div as *child* elements.
+	- After creating that , your ``App.js`` will look like this 
+	 - ```
+		import React from 'react';
+		import './App.css';
+
+		function App() {
+  			return (
+    				<div className="App">
+      					{/* Start Parent Div */}
+						  <div>
+	  						{/* Start of Child Divs */}
+							<div>Child Div 1</div>
+							<div>Child Div 2</div>
+							<div>Child Div 3</div>
+							{/* End of Child Divs */}
+	  					</div>
+	 					{/* End of Parent Div */}
+					</div>
+  				);
+		}
+
+		export default App;
+		```
+ Now we will be adding CSS to this App to differentiate **div** and achieve this one by one using
+ - Inline CSS 
+ - External CSS file and
+ - then by using Bootstrap.
+
+ To Differentiate we will add background color simple padding and dividing **parent div** into *flex* box.
+
+#### Note - To compare the changes among the 3 methods of CSS, create a separate react app.
+
+1. Using Inline CSS 
+    1. Let us give a background color and some margin on top and bottom to parent element and make its display as flex.  
+	 > ``` <div style={ { margin:"10px 0", backgroundColor:"orange", display:"flex"}} > ``` 
+    2. Now add ``background-color`` to each **child div** with some padding and flex basis.
+	> ``` <div style={{flexBasis:"25%",padding:"10px",backgroundColor:"green"}}>Child Div 1</div> ``` 
+	3. Your ``App.js`` will look like -
+
+	```
+       <div className="App">
+	   		{{/* Start of Parent Div */}}
+      		<div style={{margin:"10px 0",backgroundColor:"orange",display:"flex"}}>
+			{{/* Start of Child Divs */}}  
+			<div style={{flexBasis:"25%",padding:"10px",backgroundColor:"green"}}>Child Div 1</div>
+			<div style= {{flexBasis:"25%",padding:"10px",backgroundColor:"red"}} >Child Div 2</div>
+			<div style={{flexBasis:"25%",padding:"10px",backgroundColor:"blue"}}>Child Div 3</div>
+	  		</div>
+			{{/* End of Parent Div */}}  
+		</div>
+      ```
+2. Using External CSS 
+    1. Create an external CSS file and add a parent class with same styling as you did in Inline CSS.
+	2. Create a class for each **child element**.
+	3. Your CSS file will look like -
+		- ```
+       		.parent {
```

Comment:
please look at the markdown preview - all of these code samples' indentation is a bit weird.

Reply:
![image](https://user-images.githubusercontent.com/29131689/77262107-bf3f4680-6c50-11ea-996c-ea5e4528d813.png)